core/components/molecules/modal/Modal.tsx

```diff
@@ -36,67 +35,124 @@ export interface ModalProps extends BaseProps {
   children?: React.ReactNode;
 }
 
-const Modal = (props: ModalProps) => {
-  const {
-    dimension = 'small',
-    children,
-    backdropClose,
-    backdrop,
-    className
-  } = props;
-  const [open, setOpen] = useState<boolean>(props.open);
-  const [animate, setAnimate] = useState<boolean>(false);
-
-  const classes = classNames({
-    Modal: true,
-    [`Modal--${dimension}`]: dimension,
-    'Modal--open': open,
-    'Modal-animation--open': animate,
-    'Modal-animation--close': !animate,
-  }, className);
-
-  const baseProps = extractBaseProps(props);
-
-  useEffect(() => {
-    if (props.open) {
-      setOpen(true);
-      setAnimate(true);
+interface ModalState {
+  open: boolean;
+  animate: boolean;
+  zIndex?: number;
+}
+
+class Modal extends React.Component<ModalProps, ModalState> {
+  modalRef = React.createRef<HTMLDivElement>();
+  element: Element;
+
+  constructor(props: ModalProps) {
+    super(props);
+
+    let element = document.querySelector('.Modal-wrapper');
+    if (element === null) {
+      element = document.createElement('div');
+      element.classList.add('Modal-wrapper');
+      document.body.appendChild(element);
     }
-    if (!props.open) {
-      setTimeout(() => {
-        setOpen(false);
-      }, 120);
-      setAnimate(false);
+
+    this.element = element;
+
+    this.state = {
+      open: props.open,
+      animate: false,
+    };
+  }
+
+  componentDidUpdate(prevProps: ModalProps) {
+    if (prevProps.open !== this.props.open) {
+      if (this.props.open) {
+        const zIndex = this.getUpdatedZIndex();
+        this.setState({
+          zIndex,
+          open: true,
+          animate: true
+        });
+      } else {
+        this.setState({
+          animate: false,
+        });
+        setTimeout(() => {
+          this.setState({
+            open: false
+          });
+        }, 120);
+      }
     }
-  }, [props.open]);
+  }
+
+  getUpdatedZIndex = () => {
+    if (this.element === null) return;
+
+    const elements = this.element.querySelectorAll('.Modal-container');
+    if (elements.length <= 1) return;
+
+    const siblings = Array.from(elements).filter(el => el !== this.modalRef.current);
+    let zIndex = -1;
 
-  const ModalContainer = (
-    <div className="Modal-container">
-      <div {...baseProps} className={classes}>
-        {children}
+    siblings.forEach(element => {
+      if (element.classList.contains('Modal-container--open')) {
+        zIndex = Math.max(zIndex, parseInt(window.getComputedStyle(element).zIndex || '0', 10) + 10);
```

Comment:
`window.getComputedStyle(element).zIndex || '0'`
Store this in a variable to increase readability.